src/ui/public/query_manager/index.js

```diff
@@ -0,0 +1,79 @@
+import _ from 'lodash';
+import { FilterManagerProvider } from 'ui/filter_manager';
+import { FilterBarLibMapAndFlattenFiltersProvider } from 'ui/filter_bar/lib/map_and_flatten_filters';
+import { FilterBarLibExtractTimeFilterProvider } from 'ui/filter_bar/lib/extract_time_filter';
+import { FilterBarLibChangeTimeFilterProvider } from 'ui/filter_bar/lib/change_time_filter';
+import { toKueryExpression, fromKueryExpression, nodeTypes, filterToKueryAST } from 'ui/kuery';
+
+export function QueryManagerProvider(Private) {
+  const filterManager = Private(FilterManagerProvider);
+  const mapAndFlattenFilters = Private(FilterBarLibMapAndFlattenFiltersProvider);
+  const extractTimeFilter = Private(FilterBarLibExtractTimeFilterProvider);
+  const changeTimeFilter = Private(FilterBarLibChangeTimeFilterProvider);
+
+  return function (state) {
+
+    function add(field, values = [], operation, index) {
```

Comment:
> And put your optional arguments at the end.

-- [Kibana JavaScript Style Guide](https://github.com/elastic/kibana/blob/master/style_guides/js_style_guide.md#default-argument-syntax)

Aside from that, maybe we should take this chance to improve this function's signature? Or is it meant to be a compatibility adapter?